frontend/src/pages/inventory/Inventory.tsx

```diff
@@ -1,66 +1,165 @@
     const items = response.message as Item[];
-    setItems(items);
+    setItems(items.sort(sortItemsByType));
+    setDisplayItems(items);
+  };
+
+  const archiveItem = async (id: number) => {
+    const data = [{ id: id, archive: true }];
+
+    fetch(API_ARCHIVE_GOOD, {
+      method: 'POST',
+      headers: {
+        Authorization: `bearer ${localStorage.token}`,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(data)
+    });
+    setItems(items.filter(item => item.id !== id));
+    setDisplayItems(displayItems.filter(item => item.id !== id));
+  };
+
+  const handleClickFilter = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const filterItems = (type: number, search: string) => {
+    let itemsList = items;
+    switch (type) {
+      case 0:
+        break;
+      case 1:
+        itemsList = itemsList.filter(filterForRawItems);
+        break;
+      case 2:
+        itemsList = itemsList.filter(filterForSemiItems);
+        break;
+      case 3:
+        itemsList = itemsList.filter(filterForFinishedItems);
+        break;
+      default:
+        break;
+    }
+    setFilter(type);
+    setSearch(search);
+    setDisplayItems(
+      itemsList.filter(item => item.name.toLowerCase().includes(search.toLowerCase()))
+    );
+  };
+
+  const handleCloseFilter = (type: number) => {
+    filterItems(type, search);
+    setAnchorEl(null);
   };
 
   useEffect(() => {
     getItems();
   }, []);
 
   return (
-    <Container>
-      <div className="Inventory">
-        <div className="inventory__top">
+    <Container title="Inventory" className="Inventory">
+      <div className="inventory__top">
+        <div className="inventory__top__search">
           <h1 className="title">Summary</h1>
-          <div className="inventory__top__buttons">
-            <ImportButton />
-            <ExportButton />
-            <Button color="primary" variant="contained" component={Link} to="/inventory/add-item">
-              Add Item
-            </Button>
-          </div>
+          <InputBase
+            className="search"
+            placeholder={'Search...'}
+            startAdornment={<Search />}
+            onChange={e => filterItems(filter, e.target.value)}
+          />
+          <Button color="primary" variant="outlined" onClick={handleClickFilter}>
+            Filter by
+          </Button>
+          <Menu
+            anchorEl={anchorEl}
+            keepMounted
+            open={Boolean(anchorEl)}
+            onClose={handleCloseFilter}>
+            <MenuItem onClick={() => handleCloseFilter(0)}>{filters[0]}</MenuItem>
+            <MenuItem onClick={() => handleCloseFilter(1)}>{filters[1]}</MenuItem>
+            <MenuItem onClick={() => handleCloseFilter(2)}>{filters[2]}</MenuItem>
+            <MenuItem onClick={() => handleCloseFilter(3)}>{filters[3]}</MenuItem>
+          </Menu>
+          <p>{filters[filter]}</p>
```

Comment:
You can just use a [``<Select>``](https://material-ui.com/components/selects/#select) input and get rid of most of the logic you wrote manually.